cmd/puma-dev/main_darwin_test.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/puma/puma-dev/homedir"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainPumaDev_Darwin(t *testing.T) {
+	appLinkDir := homedir.MustExpand("~/.gotest-macos-puma-dev")
+
+	defer linkAllTestApps(t, appLinkDir)()
+
+	configureAndBootPumaDevServer(t, map[string]string{
+		"dir":        appLinkDir,
+		"dns-port":   "65053",
+		"http-port":  "65080",
+		"https-port": "65443",
+	})
+
+	runPlatformAgnosticTestScenarios(t)
```

Comment:
most examples are shared, but platform-specific stuff can live in this method back in `main_test.go`